### ISSUE TITLE
Unscented Kalman Filter: suggestions to note about required upper triangular matrix as a root

### ISSUE THREAD

**Issue by Gluttton** [rlabbe/Kalman-and-Bayesian-Filters-in-Python#91]
Recently I'm bump into your comment in the [filterpy](https://github.com/rlabbe/filterpy/blob/master/filterpy/kalman/sigma_points.py#L60) implementation of sigma point calculation:

```
If your method returns a triangular matrix it must be upper
triangular. Do not use numpy.linalg.cholesky - for historical
reasons it returns a lower triangular matrix.
```
In my opinion it will be great add note about requirement upper triangular matrix instead of lower after words: 
```
SciPy provides cholesky() method in scipy.linalg.
If your language of choice is Fortran, C, or C++,
libraries such as LAPACK provide this routine.
Matlab provides chol().
```

P.S.

Can you explain where did this requirements from?

---
I've tried use both of matrix view. When I use lower triangular matrix the covariance matrix `P` is growing up (after each step my confidence decreases).

Is it expected behaviour?

---
I try to found out how to find upper triangular matrix using Cholesky decomposition. The [most](https://en.wikipedia.org/wiki/Cholesky_decomposition#Statement) of sources say that Cholesky decomposition is:

``` A = L * L' ```

The [few](http://mathworld.wolfram.com/CholeskyDecomposition.html) sources say about another form:

``` A = U' * U ```

So it's looks like ```U = L' ```, am I right? Does the upper triangular root matrix just transposed lower triangular root matrix?

**Comment by rlabbe** [comment]
The requirement is just based on how my code is written - I wrote it to expects an upper triangular matrix. It would be trivial to change the code to use a lower triangular one instead. Upper is the default return value of scipy.linalg.cholesky, and the general advice is to prefer the scipy version for linalg over the numpy ones, so I went with that. There is no 'reason' beyond how I wrote the loop to iterate over it.

You are correct the lower/upper are just transposes of each other.

http://docs.scipy.org/doc/scipy-0.15.1/reference/generated/scipy.linalg.cholesky.html
